src/tools/gcc.jam

```diff
@@ -728,6 +728,12 @@ toolset.flags gcc.compile OPTIONS <local-visibility>global : -fvisibility=defaul
 
 toolset.flags gcc.compile.c++ OPTIONS <rtti>off : -fno-rtti ;
 toolset.flags gcc.compile.c++ OPTIONS <exception-handling>off : -fno-exceptions ;
+toolset.flags gcc.compile.c++ OPTIONS <address-sanitizer>on : -fsanitize=address ;
+toolset.flags gcc.compile.c++ OPTIONS <undefined-sanitizer>on : -fsanitize=undefined ;
+toolset.flags gcc.compile.c++ OPTIONS <thread-sanitizer>on : -fsanitize=thread ;
+toolset.flags gcc.compile.c++ OPTIONS <address-sanitizer>norecover : -fsanitize=address -fno-sanitize-recover=address ;
+toolset.flags gcc.compile.c++ OPTIONS <undefined-sanitizer>norecover : -fsanitize=undefined -fno-sanitize-recover=undefined ;
```

Comment:
It also helps to define the environment variable UBSAN_OPTIONS=print_stacktrace=1 when running with the sanitizers, at least with the undefined one; I'm not sure this applies to other ones.